app/src/main/java/com/peacecorps/pcsa/policies_glossary/GlossaryAdapter.java

```diff
@@ -0,0 +1,266 @@
+            LayoutInflater inflater = (LayoutInflater) this._context
+                    .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+            convertView = inflater.inflate(R.layout.fragment_glossary_meaning, null);
+        }
+
+        TextView txtListChild = (TextView) convertView
+                .findViewById(R.id.word_meaning);
+
+        txtListChild.setText(Html.fromHtml(childText));
+        return convertView;
+    }
+
+    @Override
+    public int getChildrenCount(int groupPosition) {
+        return this._listDataChild.get(this._listDataHeader.get(groupPosition))
+                .size();
+    }
+
+    @Override
+    public Object getGroup(int groupPosition) {
+        return this._listDataHeader.get(groupPosition);
+    }
+
+    @Override
+    public int getGroupCount() {
+        return this._listDataHeader.size();
+    }
+
+    @Override
+    public long getGroupId(int groupPosition) {
+        return groupPosition;
+    }
+
+    @Override
+    public View getGroupView(int groupPosition, boolean isExpanded,
+                             View convertView, ViewGroup parent) {
+        String headerTitle = (String) getGroup(groupPosition);
+        if (convertView == null) {
+            LayoutInflater inflater = (LayoutInflater) this._context
+                    .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+            convertView = inflater.inflate(R.layout.fragment_glossary_item, null);
+        }
+
+        TextView lblListHeader = (TextView) convertView
+                .findViewById(R.id.word_title);
+        lblListHeader.setTypeface(null, Typeface.BOLD);
+        lblListHeader.setText(headerTitle);
+
+        return convertView;
+    }
+
+    @Override
+    public boolean hasStableIds() {
+        return false;
+    }
+
+    @Override
+    public boolean isChildSelectable(int groupPosition, int childPosition) {
+        return true;
+    }
+
+    public void filter(String textEntered)
+    {
+        prepareListData(_context,_listDataHeader,_listDataChild);
+        Iterator listIt = _listDataHeader.iterator();
+        while (listIt.hasNext())
+        {
+            String next = (String) listIt.next();
+            if(next.length() < textEntered.length() || !next.toUpperCase().startsWith(textEntered.toUpperCase())){
+                listIt.remove();
+            }
+        }
+
+        for(Iterator<Map.Entry<String, List<String>>> it = _listDataChild.entrySet().iterator(); it.hasNext(); ) {
+            Map.Entry<String, List<String>> entry = it.next();
+            if(!_listDataHeader.contains(entry.getKey())) {
+                it.remove();
+            }
+        }
+        notifyDataSetChanged();
+    }
+
+    public static void prepareListData(Context context,List<String> listDataHeader,HashMap<String, List<String>> listDataChild)
+    {
+        listDataChild.clear();
+        listDataHeader.clear();
+
+        listDataHeader.add(context.getString(R.string.aggravated_sexual_assault));
```

Comment:
Can't this done by adding these strings to array.xml. Then directly taken as an array and converting it to a list using "Arrays.asList"?